c/aws-c-common/aws_byte_cursor_read_be16_harness.i

```diff
@@ -9098,7 +9098,8 @@ void aws_byte_cursor_read_common_harness() {
     ensure_byte_cursor_has_allocated_buffer_member(&cur);
     assume_abort_if_not(aws_byte_cursor_is_valid(&cur));
     assume_abort_if_not(cur.ptr);
```

Comment:
I think that cur.ptr can still point to inaccessible memory if the `__VERIFIER_nondet` in the initialization of `cur` returns value < 2 (https://github.com/sosy-lab/sv-benchmarks/blob/c17f01470e5b14a928960aeeb5e482962c421518/c/aws-c-common/aws_byte_cursor_read_be16_harness.i#L9093). Then you get heap-buffer-overflow in the memcpy here: https://github.com/sosy-lab/sv-benchmarks/blob/c17f01470e5b14a928960aeeb5e482962c421518/c/aws-c-common/aws_byte_cursor_read_be16_harness.i#L9109

Here's a test vector that shows this problem:
```
__VERIFIER_nondet_ulong:aws_byte_cursor_read_be16_harness.i:8957:35 := i64: 0
__VERIFIER_nondet_ulong:aws_byte_cursor_read_be16_harness.i:8958:21 := i64: 2
__VERIFIER_nondet_bool:aws_byte_cursor_read_be16_harness.i:261:30 :=     i8: 0
__VERIFIER_nondet_bool:aws_byte_cursor_read_be16_harness.i:261:30 :=     i8: 0
```

Reply:
I guess a similar problem is also in the other benchmarks.

Reply:
Thank you for reviewing this, I have added the constraints on `cur`.